### PR TITLE
Added proxy into docker-in-docker config

### PR DESCRIPTION
**Pull request by kse3hi**
This is neccesary to handle proxy with "base" images

### CODE REVIEW

File: .devcontainer/scripts/configure-proxies.sh [hunk 1]
```diff
@@ -95,6 +96,21 @@ if [ "${USE_PROXIES}" = "true" ]; then
...
     echo "Acquire::http::proxy \"${HTTP_PROXY}\";" >> /etc/apt/apt.conf
     echo "Acquire::https::proxy \"${HTTPS_PROXY}\";" >> /etc/apt/apt.conf
+
+    # apply proxy for docker config
+    DOCKER_CONFIG_INITIAL_CONTENT="{}"
+    mkdir -p /home/${USERNAME}/.docker
+    CONFIG_FILE=/home/$USERNAME/.docker/config.json
+    echo $CONFIG_FILE
+    if [ -s "${CONFIG_FILE}" ]; then
+        echo "File exist"
```
Comment (doosuu) [inline .devcontainer/scripts/configure-proxies.sh]:
```suggestion
        echo "Existing docker config file found!"
```

File: .devcontainer/scripts/configure-proxies.sh [hunk 2]
```diff
@@ -95,6 +96,21 @@ if [ "${USE_PROXIES}" = "true" ]; then
     # enable root user to "apt-get" via proxy
     echo "Acquire::http::proxy \"${HTTP_PROXY}\";" >> /etc/apt/apt.conf
     echo "Acquire::https::proxy \"${HTTPS_PROXY}\";" >> /etc/apt/apt.conf
+
+    # apply proxy for docker config
+    DOCKER_CONFIG_INITIAL_CONTENT="{}"
+    mkdir -p /home/${USERNAME}/.docker
```
Comment (doosuu) [inline .devcontainer/scripts/configure-proxies.sh]: I suggest we introduce a variable for `/home/${USERNAME}/.docker` and re-use that also for `CONFIG_FILE`

File: .devcontainer/scripts/configure-proxies.sh [hunk 3]
```diff
@@ -95,6 +96,21 @@ if [ "${USE_PROXIES}" = "true" ]; then
...
+    CONFIG_FILE=/home/$USERNAME/.docker/config.json
+    echo $CONFIG_FILE
+    if [ -s "${CONFIG_FILE}" ]; then
+        echo "File exist"
+        DOCKER_CONFIG_INITIAL_CONTENT=$(cat "${CONFIG_FILE}")
+    fi
+
+    DOCKER_PROXY_JSON="{\"default\": {\"httpProxy\": \"$HTTP_PROXY\", \"httpsProxy\": \"$HTTP_PROXY\", \"noProxy\": \"$NO_PROXY\"}}"
+    echo $DOCKER_PROXY_JSON
```
Comment (doosuu) [inline .devcontainer/scripts/configure-proxies.sh]: Unnecessary echo

**Review by doosuu — DISMISSED** [review]

File: .devcontainer/scripts/configure-proxies.sh [hunk 4]
```diff
@@ -95,6 +96,21 @@ if [ "${USE_PROXIES}" = "true" ]; then
     # enable root user to "apt-get" via proxy
     echo "Acquire::http::proxy \"${HTTP_PROXY}\";" >> /etc/apt/apt.conf
     echo "Acquire::https::proxy \"${HTTPS_PROXY}\";" >> /etc/apt/apt.conf
+
+    # apply proxy for docker config
+    DOCKER_CONFIG_INITIAL_CONTENT="{}"
+    DOCKER_CONFIG_FOLDER=/home/${USERNAME}/.docker
+    mkdir -p $DOCKER_CONFIG_FOLDER
+    DOCKER_CONFIG_FILE=${DOCKER_CONFIG_FOLDER}/config.json
+    echo $DOCKER_CONFIG_FILE
```
Comment (dennismeister93) [inline .devcontainer/scripts/configure-proxies.sh]: is this echo necessary?

**Review by dennismeister93 — CHANGES_REQUESTED** [review]

File: .devcontainer/scripts/configure-proxies.sh (hunk 4 above)
Comment (kse3hi) [reply .devcontainer/scripts/configure-proxies.sh]: Looks like was added for debugging

**Review by dennismeister93 — APPROVED** [review]
lgtm

**Comment by kse3hi** [comment]
https://github.com/eclipse-velocitas/vehicle-app-python-template/pull/213

**Pull request merged**